examples/nlp/text_classification_with_transformer.py

```diff
@@ -33,6 +33,12 @@ def __init__(self, embed_dim, num_heads=8):
         self.value_dense = layers.Dense(embed_dim)
         self.combine_heads = layers.Dense(embed_dim)
 
+    def get_config(self):
+        """ Required for saving/loading the model """
```

Comment:
It is technically not required. Do you get an error message?

Reply:
This is only if you save in `.h5` format. Try removing the extension and it should work.

Reply:
A lot of people save their models in `.h5` format, so won't it be better to have `get_config()`? Is there a disadvantage if it is included?

Reply:
It makes the code example longer and more cluttered, while not directly contributing to the goals of the example.

Reply:
Hi,  fchollet,  
I follow your idea to make get_config  for audio/transormer_asr.py. But I got such error: 
AttributeError: 'TokenEmbedding' object has no attribute 'maxlen'. Is there anything wrong here? Thanks. 

class TokenEmbedding(layers.Layer):
    def __init__(self, num_vocab=1000, maxlen=100, num_hid=64):
        super().__init__()
        self.emb = tf.keras.layers.Embedding(num_vocab, num_hid)
        self.pos_emb = layers.Embedding(input_dim=maxlen, output_dim=num_hid)
    def get_config(self):
        """ Required for saving/loading the model """
        config = super().get_config().copy()
        config.update(
            {
            "maxlen": self.maxlen,
            "vocab_size": self.num_vocab,
            "embed_dim": self.num_hid,
            }
        )
        return config

    def call(self, x):
        maxlen = tf.shape(x)[-1]
        x = self.emb(x)
        positions = tf.range(start=0, limit=maxlen, delta=1)
        positions = self.pos_emb(positions)
        return x + positions